src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -19,62 +20,46 @@
 public class IndexManager {
     private static final Logger logger = LoggerFactory.getLogger(IndexManager.class);
 
-    public static PineconeConnection createIndexIfNotExistsDataPlane(int dimension, String indexType) throws IOException, InterruptedException {
+    public static AbstractMap.SimpleEntry<String, Pinecone> createIndexIfNotExistsDataPlane(int dimension, String indexType) throws IOException, InterruptedException {
```

Comment:
This function now returns a tuple of `[indexName, pineconeClient]`.

Reply:
I wanted this function to be easier to work with and allow us to use `Pinecone` rather than `PineconeConnection` in consuming tests. Looks like you found the reasoning though. 👍